### ISSUE TITLE
New message do not appear if app is open

### ISSUE THREAD

**Issue by klaszlo** [deltachat/deltachat-android#27]
I tested on two phones (sony android 5.1.1, oneplus android 5.1.1), both 
application is downloaded via f-droid (v0.1.15, 2/2/2017).

I added two gmail account, enabled less secure app login inside gmail, I rebooted both phone.
The problem is, if I send a message, on the other phone it do not appear *until* I close 
delta chat, after closing in 2-5 sec and a new notification appear. 
If I leave open the delta chat's chat window, the new message never arrives.

In the gmail account the message is properly sent and received.
As a workaround it would be nice to have a manual refresh icon inside chat window.

**Comment by rsajdok** [comment]
I have the same problem. Only after restart app I can see the new messages.

**Comment by Szaman1701** [comment]
Got the same issue, Im on Huawei 8P lite with android 6.0. I can freely send a messages, but for some reason app doesn't download responses from my mailbox, despite the fact that I got them and can read them via mail app.

**Comment by ttandej** [comment]
I have the same issue,  but I have found that it only seems to affect Gmail.  I have a gnspes mail and it updates perfectly.

**Comment by r10s** [comment]
As this seems only to affect Google, this issue should be fixed by https://github.com/r10s/messenger-backend/commit/b711919abf371209aa1eeb16df085eec67ee3880 

If the problem is still present in the upcoming release (v0.1.19 or later), please re-open this issue. For now, it is closed.

**Comment by klaszlo** [comment]
after updating to v0.1.20 (f-droid repository). The exact same bug remains.
The other side do not receive my chat, and me neither hers.

Delta chat was updated on both end.

**Comment by klaszlo** [comment]
this issue can only be reopened by @r10s .

**Comment by r10s** [comment]
Okay, we will check this again; any further information are welcome.

**Comment by klaszlo** [comment]
@r10s: first comment still holds: 
- when I close the app via the android app switcher, I get the notification within a moment. 
- the message do arrive inside the gmail account. 
- the "mailbox" inside the app, is also unreliable.
- the same error on both phone (sony z1compact, android 5.1.1 factory version, oneplus android 5.1.1)

**Comment by zoff99** [comment]
+1
this is still happening on the latest playstore version

**Comment by r10s** [comment]
> this is still happening on the latest playstore version

I think you're talking about the F-Droid-version?!

**Comment by zoff99** [comment]
@r10s no, it's the playstore version. android 6.0 not using gmail as email provider.

**Comment by r10s** [comment]
> @r10s sorry, yes. f-droid version. will the f-droid version be fix aswell?

Yes. In fact, F-Droid ist the main distribution channel for the moment.

**Comment by zoff99** [comment]
ok. so now i am confused. should it be already solved in the latest f-droid version or not yet?

**Comment by r10s** [comment]
Well, I don't know _if_ it works. I only know there were _massive_ changes, esp. regarding google. I have no problems with Google and with my other testing accounts - otherwise I would fix them ;-)

(the most recent version released is 0.1.25 which is available on F-Droid, the only release channel at the moment, to be sure, no old stuff in the database, it may be an good idea to deinstall and reinstall the app (normally, this is not carry-on, but, who knows ...))

**Comment by zoff99** [comment]
well. i dont understand what you are saying exactly.
anyway, this bug is still there.

**Comment by tbart** [comment]
I can confirm this bug on lineageos 14.1 (Android 7.1.2) on a Nexus 5 using gmx.at and 0.1.32 from f-droid.org.
If I test deltachat with a second phone (exactly the same specs) it works - mostly that is (phones are either turned on or put to sleep a minute ago)
If I send a message a day later with the second phone being turned on currently or having opened deltachat recently, the messages do not show up in deltachat. They _always_ get sent, as I see them in the web mail interface.
Restarting the app always helps as messages get downloaded again.
Sometimes messages magically appear hours or even days later.
It somehow seems the IMAP connection gets dropped in the background. However, K9-Mail works perfectly using IMAP-Push.
Can you offer some debug build or is there anything we can look for in logcat to help you dissect this bug?

**Comment by r10s** [comment]
There is no official debug version, but the Delta Chat log is very detailed. You can inspect it using the usual logcat tools.

**Comment by dezifit** [comment]
Same issue with current version (v0.9.2) and provider Unitymedia. Tested with older Cyanogen 5.1.1, see attached logcat excerpt. Anything I can do or test?
[logcat.txt](https://github.com/deltachat/deltachat-android/files/1156705/logcat.txt)

**Comment by t-oster** [comment]
@dezifit is the bug still on the current versio n?

**Comment by csb0730** [comment]
@t-oster , @dezifit ,

I tested with v0.17.3.
New message appears under all conditions:
- only app open (chat list)
- I'm in another chat
- I'm in the chat where the message is intended for.

Contact Request excluded here because anyway no alert.

So I think issue can be closed

**Comment by dezifit** [comment]
I reinstalled Delta-Chat and created a test group with 2 members. I used the standard group creation notification as test message, which was send as expected to the group participant, a standard mail account.

After receiving this notification mail I used Thunderbird to reply it, but the reply isn't shown in delta-chat. I verified with the web interface of the mail account used by delta-chat and the reply is visble/waiting in the inbox (with usual subject "Re: chat: ..."). So no, if I didn't made a stupid error, it still doesn't work.

**Comment by t-oster** [comment]
could you provide the log and the mail headers of the reply email?

**Comment by t-oster** [comment]
is it shown under `Kontaktanfragen`?

**Comment by dezifit** [comment]
No, nothing shown in "Kontaktanfragen".

**Comment by testbird** [comment]
I tried to reproduce with just one phone by using a MUA and copying old emails from a chat contact (from before deltachat even existed) into the INBOX on the server and setting them as "unread", but the messages did not even appear in DC.

It worked though with a message I sent myself (appeared on the right side)

**Comment by csb0730** [comment]
Ok a short illustration of my test:
- I used an existing (single) chat for testing
- I sent three new (!) messages to the chat

I think both circumstances are important because otherwise some more difficulties are playing with into this game!

1. I made the experience that a mailserver doesn't show old messages for a client. This is not a DC's problem, more a mail server issue. But You should pay attention to that.
2. If You are trying to sent to a "group" then more difficulties in mapping messages to a group are relevant too. See the issues which are reflecting these problems.
3. See imap logging at settings->Info->Info. There You can see what (if at all) DC is doing. Maybe You see there a hint if DC sees, download or dismiss the certain message if You make the try.

I hope this helps

**Comment by testbird** [comment]
I can see that (some days?) to reasons unknown to me the IDLE interrupted appears every half hour, and would still expect the old message copy to be displayed after a while.

> I made the experience that a mailserver doesn't show old messages for a client. This is not a DC's problem, more a mail server issue.

What do you mean by "show for a client"? How could you determine that?

My old message *sent by myself*  did show up, when copied to the inbox.
Could you try copying old messages from a local archive to the server again?
(After a while, flag them to unread again, I think that may be another issue, still.)

**Comment by r10s** [comment]
i think most of these issues are targeted by the changes to the imap-jobs in v0.18.0 and newer versions.
